components/App.vue

```diff
@@ -0,0 +1,242 @@
+          Waiting for Web3... (Must be on MainNet or Rinkeby)
+        </p>
+      </div>
+    </template>
+    <BuyButton v-if="ready" />
+    <div class="info">
+      <p>
+        Ads displayed above are loaded directly from the Ethereum Blockchain.
+        This Decentralized Application (<a
+          href="https://ethereum.stackexchange.com/questions/383/what-is-a-dapp"
+          >DApp</a
+        >) does not have a traditional backend. No MVC framework, no SQL
+        database. It's just a JavaScript application served statically from
+        Github which speaks to the Ethereum blockchain using
+        <a href="https://github.com/ethereum/web3.js/">Web3.js</a>. Pretty cool,
+        right?
+      </p>
+      <p>
+        Want to see it in action?
+        <a href="https://gfycat.com/BleakSimilarGermanspaniel">Here's a GIF!</a>
+      </p>
+    </div>
+    <Footer>
+      <ul>
+        <li><h3>Blockchain</h3></li>
+        <li>
+          <Dropdown
+            :options="availableNetworks"
+            :default="activeNetwork"
+            @selected="setNetwork"
+            :disabled="!isReadOnly"
+            invalidName="Unsupported Network"
+          ></Dropdown>
+          <span v-if="!networkConfig.etherscanLink">
+            Contract is only on MainNet and Rinkeby.
+          </span>
+        </li>
+        <li v-if="networkConfig.etherscanLink">
+          <a :href="networkConfig.etherscanLink" target="_blank">
+            Contract on Etherscan
+          </a>
+        </li>
+        <li>
+          <a
+            v-if="$store.state.gridVis"
+            v-on:click="$store.commit('setVis', 'list')"
+            >List View</a
+          >
+          <a v-else v-on:click="$store.commit('setVis', 'grid')">Grid View</a>
+        </li>
+        <li v-if="$store.state.numNSFW > 0">
+          <a v-if="!showNSFW" v-on:click="showNSFW = true"
+            >Show NSFW ({{ $store.state.numNSFW }})</a
+          >
+          <a v-else v-on:click="showNSFW = false">Hide NSFW</a>
+        </li>
+      </ul>
+    </Footer>
+  </div>
+</template>
+
+<script>
+import Web3 from "web3";
+import { ethers } from "ethers";
+
+import contractJSON from "../build/contracts/KetherHomepage.json";
+
+const deployConfig = {
+  rinkeby: {
+    contractAddr: "0xb88404dd8fe4969ef67841250baef7f04f6b1a5e",
+    web3Fallback:
+      "https://rinkeby.infura.io/v3/fa9f29a052924745babfc1d119465148",
+    etherscanLink:
+      "https://rinkeby.etherscan.io/address/0xb88404dd8fe4969ef67841250baef7f04f6b1a5e",
+    prerendered: {
+      image:
+        "https://storage.googleapis.com/storage.thousandetherhomepage.com/rinkeby.png",
+      data: "https://storage.thousandetherhomepage.com/rinkeby.json",
+      loadRemoteImages: true,
+      loadFromWeb3: true,
+    },
+  },
+  main: {
+    //TODO this was called "main" in web3, but ethers calls mainnet "homestead". Will using the name homestead break during some furture eth upgrade?
```

Comment:
Yea, change it to `homestead`, but maybe keep an alias with `main` just in case.